internal/services/batch/batch_pool_resource.go

```diff
@@ -83,6 +84,19 @@ func resourceBatchPool() *pluginsdk.Resource {
 				MaxItems: 1,
 				Elem: &pluginsdk.Resource{
 					Schema: map[string]*pluginsdk.Schema{
+						// Property `node_deallocation_option` is set to be a writeOnly property by service team
+						// It can only perform on PUT operation and is not able to perform GET operation
```

Comment:
this seems like an API bug?

Reply:
I have consult with service team about this, they said this is by design a write only property and they are not going to change this.

Reply:
That seems like a weird choice but is what it is, we have an open issue to discuss that so not a blocker